django/contrib/postgres/fields/ranges.py

```diff
@@ -132,6 +133,45 @@ def db_type(self, connection):
 RangeField.register_lookup(lookups.Overlap)
 
 
+class DateTimeRangeContains(models.Lookup):
+    lookup_name = 'contains'
+    internal_type = 'DateTimeField'
+
+    def as_sql(self, compiler, connection):
+        self._pre_process_rhs(compiler)
```

Comment:
extracting that was simpler and being able to early return [here](https://github.com/django/django/pull/6885/files#diff-51fd6dc75175234f25f579ae5b7cbba3R166) without having to re-run `resolve_expression` for nothing.